File: rules/javascript/tests/test_idor_tp.js
// True positive: IDOR via direct findById without ownership check
const express = require('express');

app.get('/invoice/:id', async (req, res) => {
    // ruleid: ez-express-idor-param
    const invoice = await Invoice.findById(req.params.id);
    res.json(invoice);
});

File: rules/java/tests/test_idor_tp.java
// True positive: IDOR via direct findById without ownership check
@RestController
public class InvoiceController {

    @Autowired
    private InvoiceRepository invoiceRepo;

    @GetMapping("/invoice/{id}")
    public Invoice getInvoice(@PathVariable Long id) {
        // ruleid: ez-spring-idor-path-variable
        return invoiceRepo.findById(id).orElseThrow();
    }
}
